extensions/amp-experiment/1.0/mutation/attribute-mutation-default-url.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertAttributeMutationFormat} from './mutation';
+import {assertHttpsUrl} from '../../../../src/url';
+import {user} from '../../../../src/log';
+
+const TAG = 'amp-experiment attribute-mutation-default-url';
+
+/**
+ * Mutation for attribute (url) mutations, for both
+ * the 'href' and 'src' attributes,
+ * on unspecifed elements.
+ *
+ * @implements {./mutation.Mutation}
+ */
+export class AttributeMutationDefaultUrl {
+  /**
+   * @param {!JsonObject} mutationRecord
+   * @param {!Array<Element>} elements
+   */
+  constructor(mutationRecord, elements) {
+    this.mutationRecord_ = mutationRecord;
+    this.elements_ = elements;
+    assertAttributeMutationFormat(this.mutationRecord_);
+  }
+
+  /** @override */
+  validate() {
+    try {
+      assertHttpsUrl(
```

Comment:
I remember we agree on only allowing url change for `<amp-img>` and `<amp-iframe>`. We may also need to support `<a>`. But let's add the limitation for safety. And also tests please : )